src/rdiff_backup/rpath.py

```diff
@@ -997,8 +1000,22 @@ def get_temp_rpath(self, sibling=False):
             if self.__class__._temp_file_index > 100000000:
                 log.Log("Warning: Resetting tempfile index", 2)
                 self.__class__._temp_file_index = 0
-            tf = self.append('rdiff-backup.tmp.{index:d}'.format(
-                             index=self.__class__._temp_file_index))
+
+            if tempfile.tempdir and (shutil.disk_usage(self.path).free == 0):
+                # here to catch the bytes string passed via cli opts
+                if type(tempfile.tempdir) == bytes:
+                    tempdir = tempfile.tempdir.decode()
+                else:
+                    tempdir = tempfile.tempdir
+            else:
+                tempdir = None
+
+            if not tempdir:
+                tf = self.append('rdiff-backup.tmp.{index:d}'.format(
```

Comment:
construct the format first in a temporary variable, then use it alone or jointly with tempfile.tempdir, it avoids duplicate code.

Reply:
Addressed by https://github.com/rdiff-backup/rdiff-backup/pull/545/commits/d48390b80cf4de90f79926b8e3a3d4cdd4a2efea.